src/reviewRenderer/renderers/interactions/TextEntryInteraction.js

```diff
@@ -17,10 +17,148 @@
+};
+
+/**
+ * Set the response to the rendered interaction.
+ *
+ * The response format follows the IMS PCI recommendation :
+ * http://www.imsglobal.org/assessment/pciv1p0cf/imsPCIv1p0cf.html#_Toc353965343
+ *
+ * Available base types are defined in the QTI v2.1 information model:
+ * http://www.imsglobal.org/question/qtiv2p1/imsqti_infov2p1.html#element10333
+ *
+ * Special value: the empty object value {} resets the interaction responses
+ *
+ * @param {object} interaction
+ * @param {object} response
+ */
+const setResponse = (interaction, response) => {
+    let responseValue;
+
+    try {
+        responseValue = pciResponse.unserialize(response, interaction);
+    } catch (e) {}
+
+    if (responseValue && responseValue.length) {
+        interaction.getContainer().text(responseValue[0]);
+    }
+};
+
+/**
+ * Return the response of the rendered interaction
+ *
+ * The response format follows the IMS PCI recommendation :
+ * http://www.imsglobal.org/assessment/pciv1p0cf/imsPCIv1p0cf.html#_Toc353965343
+ *
+ * Available base types are defined in the QTI v2.1 information model:
+ * http://www.imsglobal.org/question/qtiv2p1/imsqti_infov2p1.html#element10333
+ *
+ * @param {object} interaction
+ * @returns {object}
+ */
+const getResponse = interaction => {
+    const ret = { base: {} },
+        $input = interaction.getContainer(),
+        attributes = interaction.getAttributes(),
+        baseType = interaction.getResponseDeclaration().attr('baseType'),
+        numericBase = attributes.base || 10;
+    let value;
+
+    if ($input.hasClass('invalid') || (attributes.placeholderText && $input.text() === attributes.placeholderText)) {
+        //invalid response or response equals to the placeholder text are considered empty
+        value = '';
+    } else {
+        if (baseType === 'integer') {
+            value = locale.parseInt($input.text(), numericBase);
+        } else if (baseType === 'float') {
+            value = locale.parseFloat($input.text());
+        } else if (baseType === 'string') {
+            value = $input.text();
+        }
+    }
+
+    ret.base[baseType] = isNaN(value) && typeof value === 'number' ? '' : value;
+
+    return ret;
+};
+
+const destroy = interaction => {
+    //remove all references to a cache container
+    containerHelper.reset(interaction);
+};
+
+/**
+ * Set the interaction state. It could be done anytime with any state.
+ *
+ * @param {Object} interaction - the interaction instance
+ * @param {Object} state - the interaction state
+ */
+const setState = (interaction, state) => {
+    if (_.isObject(state)) {
+        if (state.response) {
+            interaction.resetResponse();
+            interaction.setResponse(state.response);
+        }
+    }
+};
+
+/**
+ * Get the interaction state.
+ *
+ * @param {Object} interaction - the interaction instance
+ * @returns {Object} the interaction current state
+ */
+const getState = interaction => {
+    const state = {};
```

Comment:
Looks like state variable is unnecessary

Reply:
Not sure if it's a big deal, but ok